### ISSUE TITLE
ignore domain name in login window

### ISSUE THREAD

**Issue by idesk2596** [neutrinolabs/xrdp#1905]
Hi,

Can you please add support to ignore domain name in login name? 

The rdp windows client sometimes add *DOMAIN*\username in the xrdp login window
which fail the login as the pam get it as PAM_USER=DOMAIN\username
so the user always need to remove the "DOMAIN\" part.

I want xrdp to obey default pam restrictions, but it can't understand DOMAIN\ prefix.

xrdp-sesman[35456]: pam_unix(xrdp-sesman:auth): authentication failure; logname= uid=0 euid=0 tty=xrdp-sesman ruser= rhost=
xrdp-sesman[35456]: pam_krb5(xrdp-sesman:auth): authentication failure; logname=DOMAIN\username uid=0 euid=0 tty=xrdp-sesman ruser= rhost=

Thanks.

**Comment by matt335672** [comment]
Hi @idesk2596 

I'd like to get some more information on your configuration so we can work out exactly why this is happening, and what the best fix is. I've got a couple of questions for you:-
1. By 'the RDP windows client', do you mean mstsc.exe, or the Windows store client?
2. Assuming you're using mstsc.exe, normally the login box would be filled by a username if this is specified in the .ini file which is used to launch mstsc.exe. Is that the case here, or is some other mechanism filling in the username field?

Thanks

**Comment by idesk2596** [comment]
Yes, the mstsc client append the DOMAIN\ into the "username" field..
or if a user append it by mistake in the mstsc client.. 
this part in the username not relevant for Linux authentication.
(if using pam_krb5 library the domain part being calculated through /etc/krb5.conf)

i want the xrdp fron-end split the username field (delimiter by '\\') to  proceed
only with the last part which is the username, this will help to avoid such black hole cases..

adding username=asknodomain can work? (like the ask5900)

config:
[Xvnc]
##name=Xvnc
name=Remote Desktop Service
lib=libvnc.so
username=ask
password=ask
ip=127.0.0.1
port=-1
#port=ask
xserverbpp=24

![p1](https://user-images.githubusercontent.com/85168261/120339768-108ba100-c2fe-11eb-9f18-30286f7c2750.PNG)

![p2](https://user-images.githubusercontent.com/85168261/120339785-14b7be80-c2fe-11eb-90c7-695e3f766ffe.PNG)

[Xvnc]
##name=Xvnc
name=Remote Desktop Service
lib=libvnc.so
username=ask
password=ask
ip=127.0.0.1
port=-1
#port=ask
xserverbpp=24

**Comment by matt335672** [comment]
OK - thanks for the info. I think that's clear now.

**Comment by evanlinde** [comment]
The `DOMAIN\username` form has always worked for me and been resolvable via `getent passwd`, but everything I have setup uses either pam_sss or pam_winbind (not pam_krb5).

It looks like the `alt_auth_map` setting might be a helpful workaround for pam_krb5 -- something like `alt_auth_map = DOMAIN\%s`. Here are a couple reference links:

https://www.eyrie.org/~eagle/software/pam-krb5/pam-krb5.html
http://manpages.ubuntu.com/manpages/bionic/man5/pam_krb5.5.html
